dashboard-api/internal/platform/storage/study_repository.go

```diff
@@ -4,33 +4,52 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/gosimple/slug"
 	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
 )
 
 type StudyRepository struct {
 	db *sql.DB
 }
 
+// NewStudyRepository returns a new instance of the Study Repository object
+// to manage calls to the database
 func NewStudyRepository(db *sql.DB) *StudyRepository {
 	return &StudyRepository{
 		db: db,
 	}
 }
 
-func (r *StudyRepository) GetStudyBySlug(ctx context.Context, slug, userId string) (studiesmanager.Study, error) {
+func (r *StudyRepository) GetStudyBySlug(
+	ctx context.Context,
+	slug, userId string,
+) (studiesmanager.Study, error) {
 	var id, name string
 	var created time.Time
 
-	row := r.db.QueryRow("SELECT id, name, created FROM studies WHERE slug = $1 AND user_id = $2", slug, userId)
+	row := r.db.QueryRow(
+		"SELECT id, name, created FROM studies WHERE slug = $1 AND user_id = $2",
+		slug,
+		userId,
+	)
 
 	if err := row.Scan(&id, &name, &created); err != nil {
 		if err == sql.ErrNoRows {
-			return studiesmanager.Study{}, fmt.Errorf("%w: %s", studiesmanager.ErrStudyNotFound, slug)
+			return studiesmanager.Study{}, fmt.Errorf(
+				"%w: %s",
+				studiesmanager.ErrStudyNotFound,
+				slug,
+			)
 		}
 
-		return studiesmanager.Study{}, fmt.Errorf("error trying to search a study with slug '%s' on the database: %v", slug, err)
+		return studiesmanager.Study{}, fmt.Errorf(
+			"error trying to search a study with slug '%s' on the database: %v",
```

Comment:
if you are talking about the formating symbols, we have a string like such:
```go
message := "error with %s"
```
and we can then inject variables into that string dynamically:
```go
resource := "user"

fmt.Printf(message, resource)

// This will print "error with user"
```
There a lot of different symbols that you can use you can see a list [here](https://pkg.go.dev/fmt)

Otherwise the `studiesmanager.Study{}` is a "Type" that we have defined [here](https://github.com/vlab-research/vlab/blob/main/dashboard-api/internal/study.go#L17). In this case if we have an error we will return an empty Type Study. Let me know if you want me to go into this a bit deeper